meetings/working-groups/field-keyword/FK-2024-08-07.md

```diff
@@ -0,0 +1,125 @@
+
+The working group met to review the [Nullability analysis with the `field` keyword](FK-2024-08-07%20Nullability%20analysis%20with%20the%20`field`%20keyword.md) proposal to discuss the direction and come away ready to start on spec updates to present to the LDM.
+
+## Takeaways
+
+There was agreement on the changes laid out in the document. This included taking the proposed version of null-resilient getters, instead of the alternatives:
+
+- We like the version of null-resilient getters that take all nullability warnings into consideration, not just ensuring that exit points return non-null expressions. This should be very cheap to implement.
+
+- We see no reason to stop halfway and apply only `[field: MaybeNull]` without `[field: AllowNull]`. The latter would just be harder to explain to users.
+
+### Motivation takeaway
+
+For the motivation to apply nullability automatically, we felt the motivation was much stronger than simply that it is too much code to write to manually specify `[field: MaybeNull]` in a common scenario. Rather, we feel that these nullability attributes are hard for users to understand and get right. It's much greater of a conceptual hurdle than we want for how commonplace it will be to write simple lazy or defaulting scenarios.
+
+```cs
+[field: MaybeNull]
+public List<int> Prop => field ??= new();
+```
+
+### Additional NRT checking for setters
+
+The [constructor initialization](FK-2024-08-07%20Nullability%20analysis%20with%20the%20`field`%20keyword.md#constructor-initialization) section proposes that "S8618: Non-nullable property 'Prop' must contain a non-null value when exiting constructor" will be appeased by calling the property setter in the constructor, trusting that the property getter will return a non-null value after that call. This has precedent in C#.
+
+This was accepted, but a suggestion was made to check the other half of this initialization scenario. The setter could be analyzed as though it had a `MemberNotNull` attribute guaranteeing that the backing field is not null by the time it exits.
+
+Applying MemberNotNull-style analysis to the setter would provide useful warnings for two kinds of issues. One, where the field was null during the initial call to the property setter:
+
+```cs
+class C
+{
+    public C() { Prop = "..." };
+
+    public string Prop
+    {
+        get;
+        set
+        {
+            // CS8602 Dereference of a possibly null reference.
+            Console.WriteLine(field.Length);
+                              ~~~~~
+            field  = value;
+        }
+    }
+}
+```
+
+And two, where the property setter can return without initializing the backing field:
+
+```cs
+class C
+{
+    public C() { Prop = "..." };
+
+    public string Prop
+    {
+        get;
+        set
+        {
+            if (condition) field = value;
+        //  CS8774 Member 'field' must have a non-null value when exiting.
+        }
+        ~
+    }
+}
+```
+
+### Nullable backing field
```

Comment:
There's actually one point we forgot to look at: value types.

Reply:
There's a line in the null-resilient getters section explicitly saying that this feature only kicks in for properties with non-nullable, non-value types.